massa-models/src/block.rs

```diff
@@ -106,34 +124,194 @@ impl BlockId {
+        buffer.extend(serialized_content);
+        Ok(())
+    }
+
+    fn deserialize<
+        'a,
+        E: ParseError<&'a [u8]> + ContextError<&'a [u8]>,
+        DC: Deserializer<Self>,
+        U: Id,
+    >(
+        _signature_deserializer: &massa_signature::SignatureDeserializer,
+        _creator_public_key_deserializer: &massa_signature::PublicKeyDeserializer,
+        content_deserializer: &DC,
+        buffer: &'a [u8],
+    ) -> IResult<&'a [u8], Wrapped<Self, U>, E> {
+        let (rest, content) = content_deserializer.deserialize(buffer)?;
+        Ok((
+            rest,
+            Wrapped {
+                signature: content.header.signature,
+                creator_public_key: content.header.creator_public_key,
+                creator_address: content.header.creator_address,
+                thread: content.header.thread,
+                id: U::new(content.header.id.hash()),
+                content,
+                serialized_data: buffer[..buffer.len() - rest.len()].to_vec(),
+            },
+        ))
+    }
+}
+/// Serializer for `Block`
+pub struct BlockSerializer {
+    header_serializer: WrappedSerializer,
+    operation_serializer: WrappedSerializer,
+    u32_serializer: U32VarIntSerializer,
+}
+
+impl BlockSerializer {
+    /// Creates a new `BlockSerializer`
+    pub fn new() -> Self {
+        BlockSerializer {
+            header_serializer: WrappedSerializer::new(),
+            operation_serializer: WrappedSerializer::new(),
+            u32_serializer: U32VarIntSerializer::new(),
+        }
     }
+}
+
+impl Default for BlockSerializer {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Serializer<Block> for BlockSerializer {
+    fn serialize(&self, value: &Block, buffer: &mut Vec<u8>) -> Result<(), SerializeError> {
+        self.header_serializer.serialize(&value.header, buffer)?;
+        self.u32_serializer.serialize(
+            &value.operations.len().try_into().map_err(|err| {
+                SerializeError::NumberTooBig(format!("too many operations: {}", err))
+            })?,
+            buffer,
+        )?;
+        for operation in value.operations.iter() {
+            self.operation_serializer.serialize(operation, buffer)?;
+        }
+        Ok(())
+    }
+}
+
+/// Deserializer for `Block`
+pub struct BlockDeserializer {
+    header_deserializer: WrappedDeserializer<BlockHeader, BlockHeaderDeserializer>,
+    operation_deserializer: WrappedDeserializer<Operation, OperationDeserializer>,
+    u32_deserializer: U32VarIntDeserializer,
+}
+
+impl BlockDeserializer {
+    /// Creates a new `BlockDeserializer`
+    pub fn new() -> Self {
+        BlockDeserializer {
+            header_deserializer: WrappedDeserializer::new(BlockHeaderDeserializer::new()),
+            operation_deserializer: WrappedDeserializer::new(OperationDeserializer::new()),
+            u32_deserializer: U32VarIntDeserializer::new(
+                Included(0),
```

Comment:
why use 2 bounds ? why not a Range ? https://doc.rust-lang.org/std/ops/struct.Range.html

Is it to ensure that we always specify bounds ?